core/src/main/java/io/confluent/connect/storage/schema/StorageSchemaCompatibility.java

```diff
@@ -169,7 +169,9 @@ public SinkRecord project(
                projected.getKey(),
                currentValueSchema,
                projected.getValue(),
-               record.kafkaOffset()
+               record.kafkaOffset(),
+               record.timestamp(),
+               record.timestampType()
```

Comment:
Would it be better to instead call:

    record.newRecord(record.topic(), record.kafkaPartition(), currentKeySchema, projected.getKey(), currentValueSchema, projected.getValue(), record.timestamp())

This is the same, except the record's offset and typestamp type are retained automatically and thus don't have to be passed in.